trust/meson.build

```
@@ -0,0 +1,165 @@
+libtrust_data_sources = [
+  'asn1.c', 'asn1.h',
+  'base64.c', 'base64.h',
+  'pem.c', 'pem.h',
+  'oid.c', 'oid.h',
+  'utf8.c', 'utf8.h',
+  'x509.c', 'x509.h'
+]
+
+asn_sources = [
+  'basic.asn',
+  'pkix.asn',
+  'openssl.asn'
+]
+
+asn_h_deps = []
+foreach name : asn_sources
+  asn_h = custom_target('generate @0@.h'.format(name),
+                        output: '@0@.h'.format(name),
+                        input: name,
+                        command: [
+                          asn1Parser, '-o', '@OUTPUT@', '@INPUT@'
+                        ])
+  asn_h_dep = declare_dependency(sources: asn_h)
+  asn_h_deps += asn_h_dep
+endforeach
+
+libtrust_data = static_library('libtrust-data',
+                               sources: libtrust_data_sources,
+                               include_directories: [configinc, commoninc],
+                               dependencies: asn_h_deps + libtasn1_deps)
+
+libtrust_sources = [
+  'builder.c', 'builder.h',
+  'digest.c', 'digest.h',
+  'index.c', 'index.h',
+  'parser.c', 'parser.h',
+  'persist.c', 'persist.h',
+  'module.c', 'module.h',
+  'save.c', 'save.h',
+  'session.c', 'session.h',
+  'token.c', 'token.h',
+  'types.h'
+]
+
+install_data('p11-kit-trust.module', install_dir: p11_package_config_modules)
+
+p11_kit_trust_c_args = [
+  '-DP11_DEFAULT_TRUST_PREFIX="@0@"'.format(datadir),
+  '-DP11_SYSTEM_TRUST_PREFIX="@0@"'.format(sysconfdir)
+]
+
+shared_module('p11-kit-trust',
+              name_prefix: '',
+              sources: libtrust_sources + ['module-init.c'],
+              include_directories: [configinc, commoninc],
+	      c_args: p11_kit_trust_c_args,
+              dependencies: libtasn1_deps,
+              link_with: [libtrust_data, libp11_library, libp11_common],
+              install: true,
+              install_dir: p11_module_path)
+
+libtrust_testable_c_args = [
+  '-DP11_DEFAULT_TRUST_PREFIX="@0@\"'.format(join_paths(meson.current_build_dir(), 'default')),
```

There is an extra ' \' character?

```suggestion
  '-DP11_DEFAULT_TRUST_PREFIX="@0@"'.format(join_paths(meson.current_build_dir(), 'default')),
```

Also remember that since meson 0.49, you can just use '/' to join directories avoiding the use of `join_paths`.

---

Good catch.